### ISSUE TITLE
Atom 1.53.0 update failing

### ISSUE THREAD

**Issue by smrg-lm** [atom/atom#21694]
### Prerequisites

* [ x] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: <https://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode>
    * Followed all applicable steps in the debugging guide: <https://flight-manual.atom.io/hacking-atom/sections/debugging/>
    * Checked the FAQs on the message board for common solutions: <https://discuss.atom.io/c/faq>
    * Checked that your issue isn't already filed: <https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom>
    * Checked that there is not already an Atom package that provides the described functionality: <https://atom.io/packages>

### Description

After the release of Atom things started failing.

Atom started to ask for a querying password. I don't have anything configured to do that. Its a modal pop up that prevents using the gui system (KDE, ubuntu 20.04).
After choosing canceling many times it allows me to used the editor.
After some use it stops opening files or hangs (e.g. no syntax highlight).

### Steps to Reproduce

Using Atom as usual, problem started suddenly 3 days ago.

**Expected behavior:**

**Reproduces how often:**

Every day/boot.

### Versions

Kubuntu 20.04
Atom    : 1.53.0
Electron: 6.1.12
Chrome  : 76.0.3809.146
Node    : 12.4.0
apm  2.5.2
npm  6.14.8
node 12.4.0 x64
atom 1.53.0
python 3.8.5
git 2.25.1

**Comment by mfonville** [comment]
@smrg-lm could you please share your `/var/log/dpkg.log` ?
And could you please make a screenshot of the pop-up?

**Comment by smrg-lm** [comment]
Is not happening right now to take the screenshot.
[dpkg.log](https://github.com/atom/atom/files/5539215/dpkg.log)

**Comment by smrg-lm** [comment]
Just happened again (picture), is not possible to interact with KDE when that window is open. It only happens in atom.
![IMG_20201113_175257606](https://user-images.githubusercontent.com/703536/99120496-bcc39f80-25d9-11eb-90dd-f3db8a8807a7.jpg)

**Comment by smrg-lm** [comment]
And, yesterday, the message/window was different, I know because searched for that message: atom "an aplication wants to create a new query"

**Comment by mfonville** [comment]
Thank you for the extra info

@DeeDeeG I think we have this side-effect because of the updated dependencies in #21425

From `dpkg.log` I see that `gnome-keyring` was installed as dependency, and I think the login screen is related to it. Not sure why it is triggered with this specific weird behavior though.

@smrg-lm as work-around, could you try if `sudo apt remove gnome-keyring` resolves the issue for you?

**Comment by DeeDeeG** [comment]
Hmm, yeah. I don't think I ever got a chance to look closely at those added `Recommends` packages.

https://github.com/atom/atom/pull/21425/files

Can't say what would cause @smrg-lm's issue without looking at this closer.

**Comment by mfonville** [comment]
I don't think that would be the necessary here. Because the `recommends` are only soft dependencies. They are automatically pulled-in, but they can be removed without any proble.
That is also why I suggested the `sudo apt remove gnome-keyring` to see if that resolves at least his issue directly.

But for further investigation, we should check (if `gnome-keyring` really is the culprit) why the keyring is triggered when starting Atom (while it was not triggered if not installed) and why it claims not to be unlocked. And if this happens for other KDE users also, or which specific set of circumstances creates this behaviour.
(note to ourselves: keyring information is stored in `~/.local/share/keyrings/`)

**Comment by mfonville** [comment]
@smrg-lm one more thing to try:
if you don't use your keyring actually (I assume so?), could you run `rm ~/.local/share/keyrings/login.keyring` to remove the keyring when Atom is closed.
When starting Atom, it will ask to create a new keyring, if you don't enter any password, it should notify you that it will store the credentials plaintext. Accept. Next time starting Atom, it should not have the pop-up anymore. Could you verify this?

**Comment by smrg-lm** [comment]
@mfonville the workaround seems to work so far, thank you!

**Comment by smrg-lm** [comment]
@mfonville the work around seem to work so far, thank you! I will try to reinstall and see what you suggest.

**Comment by smrg-lm** [comment]
@mfonville It does what you say, it prompts again, accepting an empty password warns that by saving an empty password it will not safely encrypted, accept and it creates  "default" and "Default_keyring.keyring" files in ~/.local/share/keyrings/login.keyring.

**Comment by mfonville** [comment]
And then the pop-up does come not up anymore with when restarting of Atom?

**Comment by smrg-lm** [comment]
It didn't pop-up again, it does consistently pop-up only after deleting the file. I wasn't using it all day as I did without gnome-keyring installed but I think that behavior is consistent.

**Comment by mfonville** [comment]
Ok, then I think we are alright with our dependencies. But for users experiencing this we might have to document this somewhere. (This error apparently also well-documented on the internet, with either people recommending to install Seahorse and removing the password on that keyring, or giving the advice as I did to remove the file and set-up a new passwordless keyring)

**Comment by mfonville** [comment]
I have rechecked and reconsidered the `gnome-keyring` dependency. Whether `Recommends` should become `Suggests`, but comparing to other similar software and best-practices, I do think the `Recommends` is still the most appropriate. The issue should not occur for many users, and if they do, there are quite some tips on the internet for the cause+fix. And the best solution would be for KDE to finally implement a libsecret storage for kwallet (but this is delayed already for several years, they lack manpower) OR for distributions to set the appropriate configs to automatically unlock the keyring at login (as described at e.g. https://en.opensuse.org/GNOME_Keyring#Automatically_unlock_in_SDDM.2BKDE )

I would recommend to close this issue, and to put the attention at how KDE can add support for SecretService to Kwallet ( https://forum.kde.org/viewtopic.php?f=15&t=156925 )

**Comment by Aerijo** [comment]
I haven't confirmed, but I would imagine it's the `github` package that is probing the keychain.

**Comment by mfonville** [comment]
It are packages that use the `keytar` dependency. That includes indeed the `github` package, but e.g. also `apm`.

There is also some history on `apm` when libsecret was first implemented (when migrating from the old libgnomekeyring): https://github.com/atom/atom/pull/14269

**Comment by Aerijo** [comment]
Yes, I guess it occurring as soon as you open Atom could be caused by Atom checking for updates too, depending on if that needs a token. I'm not sure what the update check requires. It also might be telemetry.

**Comment by mfonville** [comment]
I did give it a quick look. `apm` uses keytar for Atom.io API token: https://github.com/atom/apm/blob/478f7be389cdb1bf38c6f284d7631cad04e62e91/src/auth.coffee

`github` uses keytar as a strategy: https://github.com/atom/github/blob/b7eab9ca6dcbfb11491f9dc478f2cc176b7ba693/lib/shared/keytar-strategy.js https://github.com/atom/github/blob/505223a056b4244b32a3b1a4fa0ac6f7dc49ab6c/bin/git-credential-atom.js https://github.com/atom/github/blob/14079a681ab6f5fb1ce73572b2d4b4348143c569/lib/models/github-login-model.js

I wonder if you would get the pop-up asking for credentials if setting the `ATOM_GITHUB_DISABLE_KEYTAR` environment variable as is used for "problematic CI" in https://github.com/atom/github/blob/b7eab9ca6dcbfb11491f9dc478f2cc176b7ba693/lib/shared/keytar-strategy.js#L35 . If that would stop the credential pop-up from appearing we would know that this would be the single spot where credentials are used.
